src/mesh/mesh.hpp

```diff
@@ -170,6 +170,12 @@ class Mesh {
 
   void OutputMeshStructure(const int dim, const bool dump_mesh_structure = true);
 
+  // Typically called within InitUserMeshData during problem generation
+  void EnrollUserRefinementCondition(AMRFlagFunc amrflag);
+  void EnrollUserMeshGenerator(CoordinateDirection dir, MeshGenFunc my_mg);
+  void EnrollUserExplicitSourceFunction(SrcTermFunc my_func);
+  void EnrollUserTimeStepFunction(TimeStepFunc my_func);
```

Comment:
Are these functions meaningfully hooked up at the moment? Should some of them be removed? For example, I'm not sure how source functions would fit within a `mesh` object now.

Reply:
Now that I look more into it, they might not be fully hooked up. If we decide to use these hooks, we'll hook them up in Parthenon as we start using them soon in AthenaPK. Otherwise, we might decide to remove them and go through the `StateDescriptor` objects instead.